lib/taskana-core/src/main/java/pro/taskana/task/api/TaskService.java

```diff
@@ -366,6 +376,80 @@ void forceDeleteTask(String taskId)
+   *     comment for the given taskId in the TaskComment.
+   * @throws TaskNotFoundException If the given taskId in the TaskComment does not refer to an
+   *     existing task.
+   */
+  TaskComment createTaskComment(TaskComment taskComment)
+      throws NotAuthorizedException, TaskNotFoundException, InvalidArgumentException;
+
+  /**
+   * Update a task comment.
+   *
+   * @param taskComment the task comment to be updated in the database.
+   * @return the updated task comment.
+   * @throws NotAuthorizedException If the current user has no authorization to update a task
+   *     comment.
+   * @throws ConcurrencyException if an attempt is made to update the task comment and another user.
+   *     updated it already.
+   * @throws TaskCommentNotFoundException If the given taskCommentId in the TaskComment does not
+   *     refer to an existing taskComment.
+   * @throws TaskNotFoundException If the given taskId in the TaskComment does not refer to an
+   *     existing task.
+   */
+  TaskComment updateTaskComment(TaskComment taskComment)
+      throws NotAuthorizedException, ConcurrencyException, TaskCommentNotFoundException,
+          TaskNotFoundException;
+
+  /**
+   * Deletes the task comment with the given Id.
+   *
+   * @param taskCommentId The id of the task comment to delete.
+   * @throws NotAuthorizedException If the current user has no authorization to delete a task
+   *     comment
+   * @throws InvalidArgumentException If the taskCommentId is null/empty
+   * @throws TaskCommentNotFoundException If the given taskCommentId in the TaskComment does not
+   *     refer to an existing taskComment.
+   * @throws TaskNotFoundException If the given taskId in the TaskComment does not refer to an
+   *     existing task.
+   */
+  void deleteTaskComment(String taskCommentId)
+      throws NotAuthorizedException, TaskCommentNotFoundException, TaskNotFoundException,
+          InvalidArgumentException;
+
+  /**
+   * Retrieves a task comment for a given taskCommentId.
+   *
+   * @param taskCommentId The id of the task comment which should be retrieved
+   * @throws TaskCommentNotFoundException If the given taskCommentId in the TaskComment does not
+   *     refer to an existing taskComment.
+   * @throws NotAuthorizedException If the current user has no authorization to retrieve a *
+   *     taskComment from a certain task
+   * @throws TaskNotFoundException If the given taskId in the TaskComment does not refer to an *
+   *     existing task.
+   */
+  TaskComment getTaskComment(String taskCommentId)
+      throws TaskCommentNotFoundException, NotAuthorizedException, TaskNotFoundException;
+
+  /**
+   * Retrieves a list of task comments for a given taskId.
+   *
+   * @param taskId The id of the task for which all task comments should be retrieved
+   *
+   * @throws NotAuthorizedException If the current user has no authorization to retrieve a
+   *     taskComment from a certain task
```

Comment:
last but not least please add "or if the user is not authorized to access the task".